mbed_tools_ci/tag_and_release.py

```diff
@@ -28,18 +28,20 @@ def tag_and_release(mode: CommitType) -> None:
         mode: release mode
 
     """
-    version = version_project(mode)
+    _check_credentials()
+    is_new_version, version = version_project(mode)
     logger.info(f'Current version: {version}')
     if not version:
         raise ValueError('Undefined version.')
     if mode == CommitType.DEVELOPMENT:
         return
-    _check_credentials()
-    _update_repository(mode, version)
-    _release_to_pypi()
+    _update_repository(mode, is_new_version, version)
+    if is_new_version:
+        _release_to_pypi()
 
 
-def _update_repository(mode: CommitType, version: str) -> None:
+def _update_repository(mode: CommitType, is_new_version: bool,
+                       version: str) -> None:
     git = GitWrapper()
     git.configure_for_github()
     if mode == CommitType.RELEASE:
```

Comment:
Should any of this happen if `is_new_version` is false? Why would you want to push releases for old versions?

Reply:
same version does not mean no files have changed: you may want to do a release of only .misc / .doc changes

Reply:
in that case, the version remains the same so you don't want to push to Pypi or tag but you want to generate the new documentations and push them to GitHub pages

Reply:
Hold on, so what's the difference? How could I get same version? Surely if we release a new version we want the version number bumped and the whole process should look the same? What am I missing?

Reply:
Isn't this going to be confusing though? You get a release commit on github, but there's no actual release. I'd find that odd. I wonder if we shouldn't KISS and just release a new version with updated docs/misc changes.

Reply:
I know this would happen though. I can change the commit message nevertheless when it is not a real release but just a doc change